frontend/vuejs/src/views/Tables/listCourse.vue

```diff
@@ -47,64 +62,109 @@
 
             </b-table>
             <b-modal id="modal-center" centered title="BootstrapVue">
                 <p class="my-4" v-if="course">Tên môn học: {{ course.name }} </p>
             </b-modal>
         </div>
-
-        <!--                        <div class="card-footer d-flex justify-content-end"-->
-        <!--                             :class="type === 'dark' ? 'bg-transparent': ''">-->
-        <!--                            <base-pagination total="30"></base-pagination>-->
-        <!--                        </div>-->
-
+        <div class="pagination">
+            <b-pagination
+                v-model="paginate.page"
+                :total-rows="paginate.total"
+                :per-page="paginate.perPage"
+                aria-controls="my-table"
+                @change="changePage"
+            ></b-pagination>
+        </div>
     </div>
 </template>
 <script>
+import {DEFAULT_OPTION, DEFAULT_PERPAGE} from "../../definition/constants";
+
 export default {
     name: 'projects-table',
 
     data() {
         return {
-            fields: ['name', 'description', 'is_active', 'actions'],
+            fields: [
+                {
+                    key: 'name', label: this.$i18n.t("course_screen.course.name")
+                },
+                {
+                    key: 'description', label: this.$i18n.t("course_screen.course.description")
+                },
+                {
+                    key: 'is_active', label: this.$i18n.t("course_screen.course.is_active"),
+                },
+                {
+                    key: 'actions', label: this.$i18n.t("course_screen.course.actions"),
+                }
+            ],
             data: null,
             type: '',
             filter: '',
-            course:
+            options: DEFAULT_OPTION,
+            paginate:
                 {
-                    name: "",
-                    description: "",
-                    is_active: "",
+                    perPage: DEFAULT_PERPAGE,
+                    total: '0',
+                    name: '',
+                    Page: '1'
                 },
+            course: null
 
         }
     },
     created() {
-        this.getCourse();
         this.getData();
 
     },
     methods: {
-        getCourse() {
-            this.$store.dispatch('course/getCourse')
-                .then(res => {
-                    this.data = res;
-                })
-        },
-        getData(id) {
-            this.$store.dispatch('course/getData',id).then(
+        getData() {
+            if (this.paginate.name) {
+
+                this.paginate.page = 1;
+            }
+            this.$store.dispatch('course/getData', {params: this.paginate}).then(
                 res => {
-                    this.course = res;
+                    this.data = res.data;
+                    this.paginate.perPage = res.per_page;
+                    this.paginate.total = res.total;
+
                 }
             )
```

Comment:
1. { params: this.paginate }
2. sử dụng async await khi call api
3. bỏ line trống